t/lib/OpenQA/Test/Utils.pm

```diff
@@ -595,9 +596,11 @@ sub mock_io_loop (%args) {
     my $io_loop_mock = Test::MockModule->new('Mojo::IOLoop');
     $io_loop_mock->redefine(    # avoid forking to prevent coverage analysis from slowing down the test significantly
         subprocess => sub ($io_loop, $function, $callback) {
-            my @result = eval { $function->() };
-            my $error = $@;
-            $io_loop->next_tick(sub { $callback->(undef, $error, @result) });
+            my @result;
+            try { @result = $function->() }
+            catch ($e) {
+                $io_loop->next_tick(sub { $callback->(undef, $e, @result) })
+            }
```

Comment:
This is not correct. next_tick also needs to be called in case of success as before

```suggestion
            my @result, $error;
            try { @result = $function->() }
            catch ($e) { $error = $e }
            $io_loop->next_tick(sub { $callback->(undef, $error, @result) });
```

Reply:
this is outdated. I wonder if you still want the declaration in one line, or is good to go as is